cPP_DSC.adoc

```diff
@@ -2674,243 +2617,254 @@ FCS_CKM.5.1:: The TSF shall derive cryptographic keys with [*selection*: _key ty
 +
-_In KDF-MAC-2S, if a CMAC is selected in the MAC step, then AES-128-CMAC must be selected in the KDF step, and 128 must be selected as the output key size. If HMAC is selected in the MAC step, then the same HMAC must be selected in the KDF._
+_In KDF-MAC-2S, if a CMAC is selected in the MAC step, then select AES-128-CMAC in the KDF step and select 128 as the output key size. If HMAC is selected in the MAC step, then select the same HMAC in the KDF._
 +
 _If deriving a symmetric key, select any of the above rows._
 +
-_If deriving an initialization vector, select [.underline]#KDF-CTR, KDF-FB, KDF-DPI, KDF-SHA, KDF-XOR, or KDF-ENC#._
-+
-_If deriving an authentication token, select [.underline]#KDF-CTR, KDF-FB, KDF-DPI , KDF-SHA, KDF-XOR, or KDF-ENC#._
-+
-_If deriving an authentication value, select [.underline]#KDF-CTR, KDF-FB, KDF-DPI , KDF-SHA, KDF-XOR, or KDF-ENC#._
-+
-_If deriving an HMAC key, select [.underline]#KDF-CTR, KDF-FB, KDF-DPI , KDF-SHA, KDF-XOR, or KDF-ENC#._
-+
-_If deriving a KMAC key, select [.underline]#KDF-CTR, KDF-FB, KDF-DPI , KDF-SHA, KDF-XOR, or KDF-ENC#._
-+
-_If deriving a secret IV, select [.underline]#KDF-SHA, KDF-MAC-1S, or KDF-MAC-2S#._
-+
-_If deriving a seed, select [.underline]#KDF-SHA, KDF-MAC-1S, or KDF-MAC-2S#._
+_If deriving an initialization vector, an authentication secret, HMAC key, or KMAC key, select KDF-CTR, KDF-FB, KDF-DPI, KDF-HASH, KDF-XOR, or KDF-ENC._
 +
-_The key size to be used in the HMAC falls into a range between L1 and L2 defined in ISO/IEC 10118 for the appropriate hash function (for example for SHA-256 L1 = 512, L2 =256) where L2 ≤ k ≤ L1._
+_If deriving a secret IV or seed, select KDF-HASH, KDF-MAC-1S, or KDF-MAC-2S._
 
 ==== FCS_CKM_EXT.8 Password-Based Key Derivation
 
 FCS_CKM_EXT.8 Password-Based Key Derivation
 
-FCS_CKM_EXT.8.1:: The TSF shall perform password-based key derivation functions in accordance with a specified cryptographic algorithm [_HMAC-[selection: SHA-256, SHA-384, SHA-512]_], with iteration count of _[assignment: number of iterations]_ using a randomly generated salt of length _[assignment: length of salt]_ and output cryptographic key sizes _[selection: 128, 192, 256]_ bits that meet the following standard: [NIST SP 800-132].
+FCS_CKM_EXT.8.1:: The TSF shall perform password-based key derivation functions in accordance with a specified cryptographic algorithm {empty}[_HMAC-[selection: SHA-256, SHA-384, SHA-512]_], with iteration count of _[assignment: number of iterations]_ using a randomly generated salt of length _[selection: 128, [assignment: greater than 128]]_ and output cryptographic key sizes _[selection: 128, 192, 256]_ bits that meet the following standard: [NIST SP 800-132 Section 5.3 (PBKDF2)].
```

Comment:
Is there any reason why it is limited to 256 bits?

Reply:
@slpotte

Reply:
@jvdsn is there a specific question about the KDF that you are thinking (like getting a 512 bit one for XTS)? I do get that there is no specific reason to have a limited range of sizes, but I guess the question I have right now is do you have a proposed change or are you just asking a question as to why?

Reply:
So the main reason I'm asking is because PBKDF2 can actually serve 2 purposes:
1) Derive a symmetric key to be used for whatever purpose (AES, HMAC, XTS, ...)
2) As a password hashing function, where the output "key" is actually not used as a key but just used to hide the password.

For the first use case, the output length is relevant. However, for the second use case, the output length is not relevant at all (assuming it's long enough to avoid collisions). I've seen vendors that use a 512-bit output length from PBKDF2, not as a symmetric key, but just to hash a password.

In any case, now that you mention it, our definitions for FCS_COP.1.1/KeyedHash require 160, 224, 256, 384, and 512-bit keys if you're following ISO for HMAC SHA-1, SHA-224, SHA-256, SHA-384, and SHA-512 respectively.

If I were to propose a change, I would do something like [selection: 128, 192, 256, [assignment: greater than 128]] to allow the ST author to specify other sizes as appropriate.

Reply:
I've added 512 to the key sizes in FCS_COP.1/SKC AND FCS_CKM_EXT.8.  These are the expected symmetric key sizes.

Reply:
@slpotte so is the intent of this SFR to only allow PBKDF to derive symmetric keys, rather than PBKDF for general-purpose password hashing?

Reply:
So that's an interesting thought actually. For the purposes of the DSC, it doesn't really have passwords, and so shouldn't need PBKDF for that purpose. But for the use of the DSC in a larger system (like a mobile device), having a PBKDF that can be used to obscure passwords without forcing a specific bit size may be very useful, but if we don't have an ability to set such "arbitrary" sizes, it would be limited to only the values that make sense for symmetric key generation.

That may not be needed in the catalog, but is something we may want to consider for the purposes of the DSC since we have an actual product (as opposed to a generic set of requirements) to test.

I think we can do this as a separate pull request instead of trying to bundle it into this one (which would give us more time to consider it).